### ISSUE TITLE
Rescope of Target/To branch triggers 2 RESCOPE_TO events

### ISSUE THREAD

**Issue by derekln1** [tomasbjerre/pull-request-notifier-for-stash#20]
When I update the target/to branch of an open pull request, 2 http requests are sent each with a different ${PULL_REQUEST_FROM_HASH}. One will contain the from commit of the source branch and the other will contain the previous commit of the target branch.

Is it possible to just trigger one http request with the PULL_REQUEST_FROM_HASH being the commit of the source branch?

**Comment by tomasbjerre** [comment]
Ok, just to make sure I get it =)

When pull request, source branch, is updated:
* The RESCOPED_FROM is sent once and with correct value of ${PULL_REQUEST_FROM_HASH}?

When pull request, target branch, is updated:
* Only the RESCOPED_TO is sent, it is sent 2 times and with different values of ${PULL_REQUEST_FROM_HASH}?

**Comment by derekln1** [comment]
Yes that's correct for both cases when source branch and target branch are updated.

For the case when the target branch is updated, 2 RESCOPED_TO events are sent with different values for ${PULL_REQUEST_FROM_HASH}:
1. One value will be the latest commit on the source branch
2. One value will be the previous commit/hash on the target branch.

**Comment by tomasbjerre** [comment]
Ok. As I said in #17, I think I would need to debug it a bit. But I dont have a Stash instance to run it on. And the one in the SDK does not accept pushing any more commits on my installation.

I'm pretty sure its:

 @EventListener
 public void onEvent(final PullRequestRescopedEvent e) {

Being invoked twice by Stash, but I have no idea why...

**Comment by tomasbjerre** [comment]
I have tried this on a real Stash installation now. But I cant reproduce the bug.

What does:
git  log --graph --full-history --all
look like?

What is the response of:
http://localhost:7990/stash/rest/prnfs-admin/1.0/
?

**Comment by derekln1** [comment]
Hey Tomas,

I upgraded to your latest release via the Universal Plugin Manager and I was doing a lot of testing and didn't see that issue again lately... I also actually added a fix in my Jenkins job to ignore the extra RESCOPED_TO request (with the wrong from hash).  I'll update you if I see the issue again.

The commit graph looks ok.

The response for trying that link was a 404 error "Oops, you've found a dead link".
I wonder if it's because I had a custom build and then upgraded to your official version... i only added in the http POST support in that custom build though.

thanks for the support
